pkg/fleet/installer/setup/common/setup.go

```diff
@@ -119,6 +121,15 @@ func (s *Setup) Run() (err error) {
 	if err != nil {
 		return fmt.Errorf("failed to write install info: %w", err)
 	}
+	isEMRorDataproc := s.flavor == "emr" || s.flavor == "dataproc"
+	if isEMRorDataproc {
+		// Add dd-agent to the yarn group to give the Agent permission to read the Spark log files.
```

Comment:
to avoid leaking flavors here, could you instead propagate on the setup []string ddAgentAdditional groups, and add them here